src/main/java/scmspain/karyon/restrouter/RouteHandlerProvider.java

```diff
@@ -0,0 +1,38 @@
+package scmspain.karyon.restrouter;
+
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Provider;
+import com.netflix.config.DynamicPropertyFactory;
+import com.netflix.hystrix.contrib.rxnetty.metricsstream.HystrixMetricsStreamHandler;
+import io.netty.buffer.ByteBuf;
+import io.reactivex.netty.protocol.http.server.RequestHandler;
+
+public class RouteHandlerProvider implements Provider<RequestHandler<ByteBuf, ByteBuf>> {
```

Comment:
I think the name of this provider is not descriptive enough, as it provides a HystrixMetricsStreamHandler.